google/resource_container_cluster.go

```diff
@@ -485,17 +490,26 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 		}
 	}
 
+	var zoneName string
+	locations := []string{}
+	regionName, isRegionalCluster := d.GetOk("region")
+	if !isRegionalCluster {
```

Comment:
Here's an idea- what if instead of separate "region" and "zone" fields, we just had a "location" field? Then once regional clusters have gone GA we can deprecate zone. I'm not totally set on this, but it does seem like it would simplify the logic. What do you think?

Reply:
Thought about that. But because that would increase the surface area of this change I decided against it. I'll do this if that is not too bad. If that makes sense :)

Reply:
Yup, that's fair. I do think that having it as two separate fields probably makes it a bit clearer to users what to put, so I'm fine with that.